nilearn/_utils/niimg_conversions.py

```diff
@@ -193,22 +223,45 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False,
                                     niimg.get_affine()))
             if verbose > 0:
                 print("...resampled to first nifti!")
-            
+
             from .. import image  # we avoid a circular import
             niimg = cache(image.resample_img, memory, func_memory_level=2,
                           memory_level=memory_level)(
-                              niimg,
-                              target_affine=target_affine,
-                              target_shape=target_item_shape)
+                                niimg,
+                                target_affine=target_affine,
+                                target_shape=target_item_shape)
             this_data = niimg.get_data()
 
-        data[..., cur_4d_index:cur_4d_index + size] = _to_4d(this_data)
+        data[..., cur_4d_index:cur_4d_index + size] = this_data
         cur_4d_index += size
 
     return new_img_like(first_niimg, data, target_affine)
 
 
-def check_niimgs(niimgs, accept_3d=False, return_iterator=False):
+def _iter_check_niimg_4d(niimgs):
+    affine = None
+    shape = None
+    for i, niimg in enumerate(niimgs):
+        try:
+            niimg = check_niimg_3d(niimg)
+            if affine is None:
+                affine = niimg.get_affine()
+                shape = niimg.shape
+            if not _check_fov(niimg, affine, shape):
+                raise ValueError(
+                    "Field of view of image #%d is different from reference "
+                    "FOV.\n"
+                    "Reference affine:\n%r\nImage affine:\n%r\n"
+                    "Reference shape:\n%r\nImage shape:\n%r\n"
+                    % (i, affine, niimg.get_affine(), shape,
+                       niimg.shape))
+            yield niimg
+        except TypeError as e:
+            raise TypeError("Error while loading image #%d: \n%s"
```

Comment:
This kind of thing will lose information about the provenance of the error which is very annoying if you are dropping into a debugger. Look at `git grep exc.args +=` for example in our code of how to decorate an existing exception.